compiler/fir/fir2ir/jvm-backend/src/org/jetbrains/kotlin/fir/backend/jvm/FirJvmSerializerExtension.kt

```diff
@@ -46,7 +47,7 @@ class FirJvmSerializerExtension @JvmOverloads constructor(
     state: GenerationState,
     private val irClass: IrClass,
     private val typeMapper: KotlinTypeMapperBase = state.typeMapper
-) : FirSerializerExtension() {
+) : FirSerializerExtension(BuiltInSerializerProtocol) {
```

Comment:
Looks like this will lead to unnecessary protobuf extensions (defined in `builtins.proto`) being serialized for JVM metadata.

I'd rather prefer if we replicated the hierarchy of `SerializerExtension` with two immediate subtypes `JvmSerializerExtension` and `KotlinSerializerExtensionBase`, with all "serializer protocol"-affected platform extensions as subtypes of the latter.

Reply:
Thank you for the review! I think I got the point; updated as suggested.